Cython/Compiler/ExprNodes.py

```diff
@@ -6840,6 +6788,206 @@ def can_be_used_for_function(function):
                     for assignment in entry.cf_assignments)
         return may_be_a_method
 
+class PyUnpackedMethodCallNode(PyMethodCallNode):
+    """
+    Calls a Python method, potentially unpacking the method out
+    of the PyMethod object.
+    """
+
+    def select_utility_code(self, kwargs_key_value_pairs, code):
+        # returns the function cname
+        if kwargs_key_value_pairs:
+            code.globalstate.use_utility_code(
+                UtilityCode.load_cached("PyObjectVectorCallKwBuilder", "ObjectHandling.c"))
+            return "__Pyx_Object_Vectorcall_CallFromBuilder"
+        elif self.kwdict:
+            code.globalstate.use_utility_code(
+                    UtilityCode.load_cached("PyObjectFastCall", "ObjectHandling.c"))
+            return "__Pyx_PyObject_FastCallDict"
+        else:
+            code.globalstate.use_utility_code(
+                UtilityCode.load_cached("PyObjectFastCall", "ObjectHandling.c"))
+            return "__Pyx_PyObject_FastCall"
+
+    def generate_unpacking_code(self, code, self_arg, arg_offset_cname, vectorcall_cname, function):
+        code.putln(f"{self_arg} = NULL;")
+        code.putln(f"{arg_offset_cname} = 0;")
+        code.putln(f"{vectorcall_cname} = __Pyx_PY_VECTORCALL_ARGUMENTS_OFFSET;")
```

Comment:
This can go into `generate_evaluation_code`, thus avoiding the `if not self.unpack` short circuit below that just says "well, the called asked for it but I don't feel like doing it". The three variables are created outside and just *changed* here conditionally, but they are always used outside and thus should always be initialised outside.